gnucash/register/register-gnome/combocell-gnome.c

```diff
@@ -501,6 +519,59 @@ gnc_combo_cell_set_value (ComboCell *cell, const char *str)
     gnc_basic_cell_set_value (&cell->cell, str);
 }
 
+/* This function looks through full_store for a partial match with newval and returns the first
+ match (which must be subsequently freed). It fills out box->item_list with found matches */
+static gchar*
+gnc_combo_cell_type_ahead_search(const gchar* newval, GtkListStore* full_store, PopBox *box)
+{
+    GtkTreeIter iter;
+    gboolean valid;
+    int num_found=0;
+    gchar *first_found = NULL;
+    gchar *match_str = NULL;
+    GError *gerror = NULL;
+    GMatchInfo *match_info = NULL;
+    GRegex *regex = NULL;
+    gchar *rep_str = NULL;
+    GRegex *regex0 = g_regex_new (":", 0, 0, &gerror);
+
+    // Replace ":" in newval with ".*:.*" so we can use regexp to match.
+    rep_str = g_regex_replace (regex0,newval,-1,0,".*:.*",0,&gerror);
+    // Then compile the regular expression based on rep_str.
```

Comment:
Replace the hard-coded `:` with the return value of `gnc_get_acount_separator_string()`. Users may set a different separator character (which thanks to UTF8 might be a string as far as C is concerned).

Reply:
Holy crap, thanks for reminding me, I completely forgot about that.